scripts/build.sh

```diff
@@ -0,0 +1,214 @@
+    -i|--image string                 The image path to use for the workflow (required).
+    -b|--builder-image string         Overrides the image to use for building the workflow image.
+    -h|--help                         Prints this help message.
+    -r|--runtime-image string         Overrides the image to use for running the workflow.
+    -w|--workflow-directory string    Path to the directory containing the workflow's files (the 'src' directory). Default: current directory.
+    -P|--no-persistence               Skips adding persistence configuration to the sonataflow CR.
+       --apply                        Applies the generated manifests in the current namespace.
+       --push                         Pushes the image to the registry after building
+
+Notes: 
+    1. The manifests will be in 'manifests' beside 'src'.
+    2. This script respects the 'QUARKUS_EXTENSIONS' and 'MAVEN_ARGS_APPEND' environment variables.
+EOF
+}
+
+declare -A args
+args["apply"]=""
+args["image"]=""
+args["no-persistence"]=""
+args["push"]=""
+args["workflow-directory"]="$PWD"
+args["builder-image"]=""
+args["runtime-image"]=""
+
+function parse_args {
+    while getopts ":i:b:r:w:hP-:" opt; do
+        case $opt in
+            h) usage; exit ;;
+            P) args["no-persistence"]="YES" ;;
+            i) args["image"]="$OPTARG" ;;
+            w) args["workflow-directory"]="$OPTARG" ;;
+            b) args["builder-image"]="$OPTARG" ;;
+            r) args["runtime-image"]="$OPTARG" ;;
+            -)
+                case "${OPTARG}" in
+                    help)
+                        usage; exit ;;
+                    apply)
+                        args["apply"]="YES" ;;
+                    no-persistence)
+                        args["no-persistence"]="YES" ;;
+                    push)
+                        args["push"]="YES" ;;
+                    image=*)
+                        args["image"]="${OPTARG#*=}" ;;
+                    workflow-directory=*)
+                        args["workflow-directory"]="${OPTARG#*=}" ;;
+                    builder-image=*)
+                        args["builder-image"]="${OPTARG#*=}" ;;
+                    runtime-image=*)
+                        args["runtime-image"]="${OPTARG#*=}" ;;
+                    *) log_error "Invalid option: --$OPTARG"; usage; exit 1 ;;
+                esac
+            ;;
+            \?) log_error "Invalid option: -$OPTARG"; usage; exit 2 ;;
+            :) log_error "Option -$OPTARG requires an argument."; usage; exit 3 ;;
+        esac
+    done
+
+    if [[ -z "${args["image"]:-}" ]]; then
+        log_error "Missing required flag: --image"
+        usage; exit 4
+    fi
+}
+
+function gen_manifests {
+    local res_dir_path="${args["workflow-directory"]}/src/main/resources"
+    local workflow_id
+
+    workflow_id=$(get_workflow_id "$res_dir_path")
+
+    cd "$res_dir_path"
+    log_info "Switched directory: $res_dir_path"
+
+    kn-workflow gen-manifest \
```

Comment:
@ElaiShalevRH probably not urgent, but instead of patching CRs ourselves, we can take the same approach now that the flags to set profile and image are available for use:
https://github.com/rhdhorchestrator/serverless-workflows/blob/main/scripts/gen_manifests.sh#L51-L54